code/datums/elements/strippable.dm

```diff
@@ -0,0 +1,511 @@
+/datum/strippable_item/mob_item_slot/proc/finish_unequip_mob(obj/item/item, mob/source, mob/user)
+	if (!source.drop_inv_item_on_ground(item))
+		return FALSE
+
+	if (ismob(source))
+		var/mob/sourcemob = source
+		sourcemob.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(sourcemob)] has been stripped of [item] by [key_name(user)]</font>")
+		user.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(user)] has been stripped of [key_name(sourcemob)] of [item]</font>")
+
+	// Updates speed in case stripped speed affecting item
+	source.recalculate_move_delay = TRUE
+
+/// A representation of the stripping UI
+/datum/strip_menu
+	/// The owner who has the element /datum/element/strippable
+	var/atom/movable/owner
+
+	/// The strippable element itself
+	var/datum/element/strippable/strippable
+
+	/// A lazy list of user mobs to a list of strip menu keys that they're interacting with
+	var/list/interactions
+
+/datum/strip_menu/New(atom/movable/owner, datum/element/strippable/strippable)
+	. = ..()
+	src.owner = owner
+	src.strippable = strippable
+
+/datum/strip_menu/Destroy()
+	owner = null
+	strippable = null
+
+	return ..()
+
+/datum/strip_menu/tgui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "StripMenu")
+		ui.open()
+
+
+/datum/strip_menu/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/simple/inventory),
+	)
+
+/datum/strip_menu/ui_data(mob/user)
+	var/list/data = list()
+
+	var/list/items = list()
+
+	for (var/strippable_key in strippable.items)
+		var/datum/strippable_item/item_data = strippable.items[strippable_key]
+
+		if (!item_data.should_show(owner, user))
+			continue
+
+		var/list/result
+
+		if(strippable_key in LAZYACCESS(interactions, user))
+			LAZYSET(result, "interacting", TRUE)
+
+		var/obscuring = item_data.get_obscuring(owner)
+		if (obscuring != STRIPPABLE_OBSCURING_NONE)
+			LAZYSET(result, "obscured", obscuring)
+			items[strippable_key] = result
+			continue
+
+		var/obj/item/item = item_data.get_item(owner)
+		if (isnull(item))
+			items[strippable_key] = result
+			continue
+
+		LAZYINITLIST(result)
+
+		result["icon"] = icon2base64(icon(item.icon, item.icon_state, frame = 1))
+		result["name"] = item.name
+		result["alternate"] = item_data.get_alternate_action(owner, user)
+
+		items[strippable_key] = result
+
+	data["items"] = items
+
+	// While most `\the`s are implicit, this one is not.
+	// In this case, `\The` would otherwise be used.
+	// This doesn't match with what it's used for, which is to say "Stripping the alien drone",
+	// as opposed to "Stripping The alien drone".
+	// Human names will still show without "the", as they are proper nouns.
+	data["name"] = "\the [owner]"
+
+	return data
+
+/datum/strip_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if (.)
+		return
+
+	. = TRUE
+
+	var/mob/user = usr
+
+	switch (action)
+		if ("use")
```

Comment:
Would it not be possible to refactor this into a both a strip / equip action? I'm thinking the code might be clearer if both options weren't handled in the same action.

Reply:
Done